docs/design/otlp.md

```diff
@@ -0,0 +1,171 @@
+     ├── http/           # Reqwest blocking/async clients, body encoder
+     └── tonic/          # Tonic clients, TLS/compression helpers
+```
+### 3.1 Common Interfaces
+* **`HasExportConfig` / `WithExportConfig`** – traits mixed into builders to expose shared config (`endpoint`, `timeout`, `protocol`).
+* **Builder marker types** (`NoExporterBuilderSet`, `HttpExporterBuilderSet`, `TonicExporterBuilderSet`) enforce at compile time that exactly *one* transport is chosen.
+* **`SupportedTransportClient` enum** – run-time dispatch inside the exporter when sending.
+
+---
+
+## 4. Feature-Flag Matrix
+| Cargo feature | Purpose | Conditional modules |
+|---------------|---------|---------------------|
+| `trace` / `metrics` / `logs` | Enable signal exporters | `span.rs`, `metric.rs`, `logs.rs` |
+| `grpc-tonic` | Use `tonic` gRPC transport | `exporter/tonic` |
+| `http-proto` *(default)* | HTTP + protobuf body | `exporter/http` |
+| `http-json` | HTTP + JSON body | `exporter/http` |
+| `gzip-tonic` `zstd-tonic` | gRPC message compression | `exporter/tonic` |
+| `reqwest-client` / `reqwest-blocking-client` *(default)* | Choose async vs blocking Reqwest HTTP client |
+| `hyper-client` | Use Hyper HTTP transport *(requires both Reqwest features disabled)* |
+| TLS helpers (`tls-roots`, `tls-webpki-roots`, `reqwest-rustls`) | Supply trust roots for TLS |
+
+Because **only one transport is valid per exporter**, `protocol` is currently a *hint* – unsupported values are ignored by the concrete builder.
+
+### 4.1 Transport-specific Client Options
+
+Exporters must function both in binaries that start no async runtime and in services already running Tokio.  Consequently the crate provides **blocking** and **non-blocking** HTTP clients behind mutually-exclusive feature-flags.  The blocking variant is the default to ensure out-of-the-box operability; applications that already depend on Tokio can opt into an async client to avoid spawning extra threads.
+
+| Implementation | Feature flag | Blocking? | Underlying crate | Transports | Typical use case |
+|---------------|--------------|-----------|------------------|-----------|------------------|
+| Reqwest (blocking) **(default)** | `reqwest-blocking-client` | Yes | `reqwest::blocking` | HTTP/1.1 → auto-upgrades to HTTP/2 over TLS | CLI tools, synchronous binaries, or hybrid apps where introducing Tokio is undesirable. A helper thread is spawned once at builder time.
+| Reqwest (async) | `reqwest-client` | No (Tokio) | `reqwest` | HTTP/1.1 → auto-upgrades to HTTP/2 over TLS | Services that already run a Tokio runtime and prefer fully non-blocking exports.
+| Hyper | `hyper-client` | No (Tokio) | `hyper` | HTTP/1.1 → auto-upgrades to HTTP/2 over TLS | Lean dependency footprint when both Reqwest features are disabled.
+| Tonic (gRPC) | `grpc-tonic` | No (Tokio) | `tonic` | HTTP/2 (gRPC) | Environments standardising on gRPC or collectors exposing only port 4317.
```

Comment:
The "typical use case" header is not adding value IMHO. If a user want to use gRPC, then they use tonic/gRPC.